src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -210,84 +209,73 @@ public boolean test(final GATKRead read){
     private static final class SVCustomReadFilter implements ReadFilter{
         private static final long serialVersionUID = 1L;
 
-        private final ReadFilter collapsedFilter;
+        private final ReadFilter combinedReadFilter;
 
-        public SVCustomReadFilter(final UseEnd whichEnd,
-                                  final boolean filterNonProperlyPairdReads,
-                                  final boolean filterUnmappedReads,
-                                  final boolean filterMateUnmappedReads,
-                                  final boolean filterDupReads,
+        public SVCustomReadFilter(final EndToUse whichEnd,
+                                  final boolean filterNonProperlyPairedReads,
+                                  final boolean filterDuplicatedReads,
                                   final boolean filterSecondaryAlignments,
                                   final boolean filterSupplementaryAlignments,
-                                  final int MQThreshold){
-
-                final UseEnd endVal = whichEnd;
+                                  final int     MQThreshold,
+                                  final int     mMQThreshold){
```

Comment:
On second thought, this might actually introduce a bug that biases result when combined with the read's own mapping quality threshold and the option about which end of a pair to use for calculating the metrics. Will discuss with you tomorrow.